src/ocean/task/TaskPool.d

```diff
@@ -181,6 +185,9 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
         foreach (task; tasks_iterator)
         {
+            enforce(!isSame(this.toItem(current_task), this.toItem(task)),
```

Comment:
Sure, I don't mind this, nor I will insist on `assert` here. I was just under impression that this check is very likely to fail the assertion during development time, where it can be safely removed from the production.